internal/display/bufwindow.go

```diff
@@ -489,6 +489,23 @@ func (w *BufWindow) displayBuffer() {
 		bloc.X = bslice
 
 		draw := func(r rune, combc []rune, style tcell.Style, highlight bool, showcursor bool) {
+			if b.Settings["altlinecolors"].(bool) {
+				// altlinecolors: apply alt line color on symbols bg
+				if (vloc.Y + w.StartLine.Line)%2 == 0 && !currentLine {
```

Comment:
1. `!currentLine` check is not needed, as long as we make sure that the style for `cursorline` is updated *after* updating style for `altlinecolors` (so that the `cursor-line` highlighting overrides the `alt-line` highlighting).

Moreover, this `!currentLine` check causes a bug: if `cursorline` option is disabled, i.e. when the current line should be highlighted with `alt-line` instead of `cursor-line`, it is not highlighted. Moreover, since we only check `!currentLine` here, and not check it when applying altlinecolors to the trailing empty part of the line, the result is: the trailing empty part is highlighted and the rest of the line is not, which makes it look even more weird.

Furthermore, as I understand, the `selectionOnLine` logic below addresses precisely the same issue, but only for the case when there is a selection (so it does not properly address the issue, rather works around it, and also makes the code even more complicated and confusing). So if we remove the `!currentLine` check, I think we can completely remove the `selectionOnLine` stuff as well.

2. Instead of `vloc.Y + w.StartLine.Line` it's better to use `bloc.Y`. It makes the code not just more readable but also IMO more correct: if `softwrap` is enabled, some lines may be wrapped, and then `vloc.Y + w.StartLine.Line` identifies a visual line, while `bloc.Y` identifies a logical line in the buffer, which may span multiple visual lines. IMO altlinecolors should highlight logical lines, not visual lines.